Formula/etcd.rb

```diff
@@ -26,25 +26,16 @@ def install
     prefix.install_metafiles
   end
 
-  def plist_unsupported_platformenv
-    on_macos do
-      if Hardware::CPU.arm?
-        # etcd isn't officially supported on arm64
-        # https://github.com/etcd-io/etcd/issues/10318
-        # https://github.com/etcd-io/etcd/issues/10677
-        return "
-          <key>EnvironmentVariables</key>
-          <dict>
-            <key>ETCD_UNSUPPORTED_ARCH</key>
-            <string>arm64</string>
-          </dict>
-          "
-      end
-    end
-  end
-
   plist_options manual: "etcd"
 
+  service do
+    environment_variables ETCD_UNSUPPORTED_ARCH: "arm64" if Hardware::CPU.arm?
+    run [opt_bin/"etcd"]
+    run_type :immediate
+    keep_alive true
+    working_dir var
+  end
+
   def plist
```

Comment:
You need to delete the `plist` method when using a `service` block.

Reply:
Removed plist section
Tested out service and it seems to work:"

```
% /opt/homebrew/bin/etcdctl put foo bar
OK
% /opt/homebrew/bin/etcdctl get foo    
foo
bar
```

Reply:
actually, not happy with that.

The problem is, the way it is, we're not supporting arm at all, and I only looked specifically at arm on macOS.
updating to catch that specific.

Reply:
Homebrew doesn't support arm in Linux so that would be fine like this